express/components/hf_datasets_components.py

```diff
@@ -0,0 +1,131 @@
+"""Hugging Face Datasets single component module """
+
+import os
+import importlib
+import tempfile
+from abc import ABC, abstractmethod
+from typing import List, Optional, Dict, Union
+
+import datasets
+from datasets import load_dataset
+
+from .common import ExpressDatasetHandler, ExpressDataset, ExpressTransformComponent, \
+    ExpressDatasetDraft, ExpressLoaderComponent
+from express.storage_interface import StorageHandlerModule
+from express.manifest import DataManifest, DataSource, DataType
+
+# Define interface of pandas draft
+# pylint: disable=unsubscriptable-object
+HFDatasetsDatasetDraft = ExpressDatasetDraft[List[str], Union[pd.DataFrame, pd.Series]]
+
+# pylint: disable=no-member
+STORAGE_MODULE_PATH = StorageHandlerModule().to_dict()[os.environ.get('CLOUD_ENV', 'GCP')]
+STORAGE_HANDLER = importlib.import_module(STORAGE_MODULE_PATH).StorageHandler()
+
+
+# pylint: disable=too-few-public-methods
+class HFDatasetsDataset(ExpressDataset[List[str], Union[datasets.Dataset, datasets.DatasetDict]]):
+    """Hugging Face Datasets dataset"""
+
+    def load_index(self) -> datasets.Dataset:
+        """Function that loads in the index"""
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            local_parquet_path = STORAGE_HANDLER.copy_file(self.manifest.index.location, tmp_dir)
+
+            dataset = load_dataset("parquet", data_files=local_parquet_path)
+
+            return dataset
+
+    @staticmethod
+    def _load_data_source(data_source: DataSource,
+                          index_filter: Union[datasets.Dataset, datasets.DatasetDict, List[str]]) -> Union[datasets.Dataset, datasets.DatasetDict]:
+        """Function that loads in a data source"""
+        if data_source.type != DataType.parquet:
+            raise TypeError("Only reading from parquet is currently supported.")
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+
+            data_source_location = data_source.location
+
+            local_parquet_path = STORAGE_HANDLER.copy_parquet(
+                data_source_location,
+                tmp_dir
+            )
+
+            data_source_hf_datasets = load_dataset("parquet", data_dir=local_parquet_path, split="train")
```

Comment:
load_dataset also supports things like multiprocessing (if there are multiple CPU cores) and streaming, not sure if we want to use that too